lib/internal/streams/pipelinify.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const pipeline = require('internal/streams/pipeline');
+const Duplex = require('internal/streams/duplex');
+const { destroyer } = require('internal/streams/destroy');
+
+module.exports = function pipe(...streams) {
+  // TODO (ronag): Avoid double buffering.
+  // Implement Writable/Readable/Duplex traits.
```

Comment:
I have some ideas in regards to creating a `duplexify` helper with close to 0 overhead.

Reply:
e.g. continue work on https://github.com/nodejs/node/pull/33515